### ISSUE TITLE
lint: add inspection for uncalled function

### ISSUE THREAD

**Issue by KotlinIsland** [denoland/deno_lint#638]
```ts
const a = {
  foo() { console.log("hi") },
  get bar() {
    console.log("side effect in bar");
    return () => { console.log("hi") };
  }
}

a.foo // should warn that this is an uncalled function reference
a.bar // even though this has a side effect, should still warn that this is an uncalled function reference
```
